compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -846,14 +846,21 @@ trait Checking {
   }
 
   /** Check that all case classes that extend `scala.Enum` are `enum` cases */
-  def checkEnum(cdef: untpd.TypeDef, cls: Symbol)(implicit ctx: Context): Unit = {
+  def checkEnum(cdef: untpd.TypeDef, cls: Symbol, firstParent: Symbol)(implicit ctx: Context): Unit = {
     import untpd.modsDeco
     def isEnumAnonCls =
       cls.isAnonymousClass &&
       cls.owner.isTerm &&
       (cls.owner.flagsUNSAFE.is(Case) || cls.owner.name == nme.DOLLAR_NEW)
-    if (!cdef.mods.isEnumCase && !isEnumAnonCls)
-      ctx.error(em"normal case $cls in ${cls.owner} cannot extend an enum", cdef.pos)
+    if (!cdef.mods.isEnumCase && !isEnumAnonCls) {
+      if (cls.is(Case))
+        ctx.error(em"normal case $cls in ${cls.owner} cannot extend an enum", cdef.pos)
+      else if (firstParent != defn.EnumClass && firstParent.derivesFrom(defn.EnumClass) && !cdef.mods.isEnumCase && !isEnumAnonCls)
+        // Since enums are classes and Namer checks that classes don't extend multiple classes, we only check the class
+        // parent.
+        //Tricky to phrase; language taken from "case-to-case inheritance is prohibited".
+        ctx.error(s"${cls.name} has enum ancestor ${firstParent.name}, but inheriting from enums is prohibited", cdef.pos)
```

Comment:
Why not simply: `${cls.name} inherits from enum ${firstParent.name}, ...`?

Reply:
Because that doesn’t say that inheriting from an enum is a problem. We explain that even for case classes, as the “irrelevant” comment points out.

Reply:
I didn't mean to remove the part after the comma, just rephrase the first part:
```scala
"${cls.name} inherits from enum ${firstParent.name}, but inheriting from enums is prohibited"
```
I find it weird to talk about ancestor when it really is the first parent. Ideally enum should be treated as effectively final and we should report errors similarly to `final` inheritance rather than case to case inheritance. Ideally something like:
```scala
enum Foo { case F }
class Bar extends Foo
class Bat extends Bar
```
shoud report similar errors as:
```scala
final class Foo
class Bar extends Foo // error: class Bar cannot extend final class Foo
class Bat extends Bar // no error here
```

Reply:
Memo to self: using the flag would not be transitive, so it would also avoid this double reporting.